### ISSUE TITLE
F/T in chunk option can produce error `object of type 'symbol' is not subsettable`. 

### ISSUE THREAD

**Issue by rubenarslan** [yihui/knitr#1324]
Calling rmarkdown::render on

    ```{r warning=F}
    ```

occasionally produces the error `object of type 'symbol' is not subsettable`. 
This never happened in a clean session, although knitr does show

```
label: unnamed-chunk-20 (with options) 
List of 1
 $ warning: symbol F
```

in a clean session too. Unfortunately I'm not able to reproduce the conditions under which this subsetting error occurs and I'm not even clear what "symbol" means in this context (it's not the class() or typeof() of F). 
Changing all my chunk options to FALSE/TRUE fixes it when the problem occurs, but I'm fairly sure I haven't overwritten F anywhere.

I'm sorry for posting a non-reproducible example, but I thought it might be helpful if others who encounter this message find this via Google, so they know how to fix it. If I manage to reproduce it, I'll post back here.

**Comment by yihui** [comment]
Here is why `F` (or `T`) is a symbol (see `?is.symbol` to know what a symbol is):

```
> str(alist(warning = F))
List of 1
 $ warning: symbol F
> str(alist(warning = FALSE))
List of 1
 $ warning: logi FALSE
```

Yes, please let me know when you are able to reproduce the problem. Without a reproducible example, I cannot fix it. Thanks!

**Comment by rubenarslan** [comment]
Ok, so I hit it again today. It appears to be unrelated to knitr and I didn't find an explanation..

A simple test document is sufficient, I called `rmarkdown::render("test_knitr.Rmd")` on it.

    ```{r cache=F}
    1
    ```

The error also occurs when I simply call `str(alist(warning = F))`, but not if I leave out `str`. If I call str and class on F and FALSE in my workspace, they seem identical for all means and purposes.

and I had the following 

## search path
>  [1] ".GlobalEnv"          "package:pander"      "package:broom"       "package:dtplyr"     
 [5] "package:dplyr"       "package:data.table"  "package:knitr"       "package:plyr"       
 [9] "package:reshape2"    "package:stringi"     "package:stringr"     "package:lubridate"  
[13] "package:formr"       "package:QuantPsyc"   "package:MASS"        "package:boot"       
[17] "package:psych"       "package:car"         "package:Hmisc"       "package:Formula"    
[21] "package:survival"    "package:lattice"     "package:foreign"     "package:formatR"    
[25] "package:rmarkdown"   "package:brms"        "package:rstan"       "package:StanHeaders"
[29] "package:ggplot2"     "tools:rstudio"       "package:stats"       "package:graphics"   
[33] "package:grDevices"   "package:utils"       "package:datasets"    "package:methods"    
[37] "Autoloads"           "package:base"       

## Session info
> R version 3.3.2 (2016-10-31)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X El Capitan 10.11.6

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pander_0.6.0         broom_0.4.1.9000     dtplyr_0.0.1         dplyr_0.5.0         
 [5] data.table_1.9.6     knitr_1.15.1         plyr_1.8.4           reshape2_1.4.2      
 [9] stringi_1.1.2        stringr_1.1.0        lubridate_1.6.0      formr_0.4.2         
[13] QuantPsyc_1.5        MASS_7.3-45          boot_1.3-18          psych_1.6.9         
[17] car_2.1-3            Hmisc_4.0-0          Formula_1.2-1        survival_2.40-1     
[21] lattice_0.20-34      foreign_0.8-67       formatR_1.4          rmarkdown_1.2       
[25] brms_1.1.0.9000      rstan_2.12.1         StanHeaders_2.12.0-1 ggplot2_2.2.0       

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.8         tidyr_0.6.0         assertthat_0.1      rprojroot_1.1       digest_0.6.10      
 [6] packrat_0.4.8-1     R6_2.2.0            chron_2.3-47        CircStats_0.2-4     backports_1.0.4    
[11] acepack_1.4.1       MatrixModels_0.4-1  stats4_3.3.2        evaluate_0.10       coda_0.18-1        
[16] highr_0.6           lazyeval_0.2.0      minqa_1.2.4         SparseM_1.74        nloptr_1.0.4       
[21] rpart_4.1-10        Matrix_1.2-7.1      labeling_0.3        splines_3.3.2       lme4_1.1-12        
[26] statmod_1.4.27      munsell_0.4.3       mnormt_1.5-5        mgcv_1.8-16         htmltools_0.3.5    
[31] nnet_7.3-12         tibble_1.2          gridExtra_2.2.1     htmlTable_1.7       grid_3.3.2         
[36] nlme_3.1-128        gtable_0.2.0        DBI_0.5-1           magrittr_1.5        scales_0.4.1       
[41] latticeExtra_0.6-28 RColorBrewer_1.1-2  tools_3.3.2         rsconnect_0.6       abind_1.4-5        
[46] parallel_3.3.2      pbkrtest_0.4-6      yaml_2.1.14         inline_0.3.14       colorspace_1.3-1   
[51] cluster_2.0.5       quantreg_5.29

**Comment by yihui** [comment]
Could you provide a minimal (avoid loading so many packages), self-contained, and reproducible example? Please also let me know how to reproduce it.

**Comment by rubenarslan** [comment]
I cannot reliably reproduce it, sorry, that's what I meant by "I didn't find an explanation". 
I tried detaching all packages (didn't unload all, but most) and cleaning the workspace in the same session and the error still occurred. I saved my workspace before and doing so. Upon restarting the session the error did not recur and I couldn't reproduce it by loading my saved workspace.

I've no further idea how to reproduce it. Usually when I have hard to debug/reproduce errors like this, it's something about the package load order, that's why I showed it.
